lib/navigation/view.ts

```diff
@@ -70,6 +70,12 @@ interface ViewData {
 	 * haven't customized their sorting column
 	 */
 	defaultSortKey?: string
+
+	/**
+	 * Method called when the view navigation entry is toggled
+	 */
+	// eslint-disable-next-line no-use-before-define
+	onToggleOpen?: (open: boolean, view: View) => Promise<void>
```

Comment:
Is the `view` param needed? Is it not "this"?

Reply:
I had the same thought this morning, but this is the callback, not what you have to provide :wink: 
Your function will be called with `open: boolean, view: View`

Reply:
(without, it would remove the use before define and does not add any new information. But this is not a blocker for me :) )

Reply:
Ah right! :facepalm: 

Indeed, this is unnecessary :)

Reply:
Also had the same thought but didn't test and thought being explicit might be nicer

Will try it...

Reply:
Don't know why but did not work :(, will go with view param then